### PR TITLE
Avoid errors on cyclic dependency of block-bundle 3 and admin-bundle 3

### PR DESCRIPTION
**Pull request by jordisala1991**
This only happens on block bundle 3, the version 4 removed the admin-bundle dependency

Proof PR: https://github.com/sonata-project/SonataBlockBundle/pull/737

### CODE REVIEW

File: templates/project/.github/workflows/test.yaml.twig [hunk 1]
```diff
@@ -95,6 +95,12 @@ jobs:
         if: matrix.variant != 'normal'
         {% verbatim %}run: composer require ${{ matrix.variant }} --no-update{% endverbatim %}
 
+{# Remove when removing 3.x for the managed branches of block-bundle #}
+{% if repository_name == 'SonataBlockBundle' and '3.x' == current_branch %}
+      - name: Allow cyclic dependency between block-bundle and admin-bundle
+        run: sed --in-place "s/\"dev-master\":/\"dev-${GITHUB_SHA}\":/" composer.json
```
Comment (jordisala1991) [inline templates/project/.github/workflows/test.yaml.twig:101]: This code was on travis, but for all bundles. IMO it is only needed on block bundle version 3: https://github.com/sonata-project/dev-kit/blob/master/templates/project/.travis/before_install_test.sh.twig#L22

Comment (jordisala1991) [reply templates/project/.github/workflows/test.yaml.twig:101]: The dependency is only cyclic when developing the bundles, for the end users is not cyclic (is on require-dev of block-bundle and require of admin-bundle)

**Review by OskarStark — APPROVED** [review]

**Review by OskarStark — DISMISSED** [review]

File: templates/project/.github/workflows/test.yaml.twig [hunk 2]
```diff
@@ -95,6 +95,12 @@ jobs:
         if: matrix.variant != 'normal'
         {% verbatim %}run: composer require ${{ matrix.variant }} --no-update{% endverbatim %}
 
+{# Remove when removing 3.x for the managed branches of block-bundle #}
+{% if repository_name == 'SonataBlockBundle' and '3.x' == current_branch %}
```
Comment (greg0ire) [inline templates/project/.github/workflows/test.yaml.twig:99]:
```suggestion
{% if repository_name == 'SonataBlockBundle' and current_branch == '3.x' %}
```

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

**Pull request merged**